src/java/com/rapleaf/jack/queries/WhereClause.java

```diff
@@ -0,0 +1,101 @@
+package com.rapleaf.jack.queries;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.Sets;
+
+import java.util.*;
+
+public class WhereClause {
+  List<WhereConstraint> whereConstraints;
+  Optional<Set<Long>> selectedIds;
```

Comment:
Could you get rid of the Optional here? It was not super useful in the first place, and since you're refactoring the `WhereClause` we might as well remove it now